### ISSUE TITLE
Blend/masked assignment primitive

### ISSUE THREAD

**Issue by ibaned** [mattkretz/std-simd-feedback#17]
Based on your comment [here](https://github.com/mattkretz/std-simd-feedback/issues/3#issuecomment-603097158), you would be open to some kind of function:
```cpp
simd<T> cond(simd_mask<T>, simd<T>, simd<T>)
```

This is consistent with APIs I've seen in several other efforts to develop a SIMD abstraction layer, and ~might be~ is more intuitive to our users.

As you mention, if the standard provides overloading of the ternary operator, this could build on that capability, however I would recommend that a named non-member method still be provided so that implementations of `std::simd` could be developed that are usable in currently available C++ standards. In other words, take advantage of but don't rely solely on language changes.

In [our efforts](https://github.com/kokkos/simd-math) to implement something close to `std::simd`, we have chosen to only provide this primitive, which we arbitrarily name `choose`.

@crtrott @dalg24

**Comment by mattkretz** [comment]
Right. I think this would maybe have been useful in the TS. For the IS merger I'll propose to either overload `operator?:` or provide the function. In terms of C++ evolution the argument to support implementations relying on older standards is rather weak.

**Comment by ibaned** [comment]
Sounds good! Either approach would provide us a good transition. I understand that the ISO C++ standard library wouldn't want to maintain both if `operator?:` can be overloaded by the language. In terms of user application evolution, we can start with a function today and use whichever approach becomes available in future C++ standards, knowing that it will have the same behavior.
